ARTstor.js

```diff
@@ -1,131 +1,484 @@
-	"minVersion": "3.1",
+	"creator": "John Justin, Charles Zeng",
+	"target": "\\.artstor|\\.sscommons\\.org:?\\w*\\/(open)*library",
+	"minVersion": "3.0",
 	"maxVersion": "",
 	"priority": 100,
 	"inRepository": true,
 	"translatorType": 4,
 	"browserSupport": "gcs",
-	"lastUpdated": "2014-11-13 15:09:17"
-}
-
-/*
-	***** BEGIN LICENSE BLOCK *****
-	
-	ARTstor Translator, Copyright © 2012 Sebastian Karcher
-	This file is part of Zotero.
-	
-	Zotero is free software: you can redistribute it and/or modify
-	it under the terms of the GNU Affero General Public License as published by
-	the Free Software Foundation, either version 3 of the License, or
-	(at your option) any later version.
-	
-	Zotero is distributed in the hope that it will be useful,
-	but WITHOUT ANY WARRANTY; without even the implied warranty of
-	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-	GNU Affero General Public License for more details.
-	
-	You should have received a copy of the GNU Affero General Public License
-	along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
-	
-	***** END LICENSE BLOCK *****
-*/
-
-function getData(doc, checkOnly) {
-	var data = {},
-		widgets = doc.getElementsByClassName('MetaDataWidget'),
-		topWidget = null, topZIndex;
-	
-	// Dtermine the top metadata window
-	for (var i=0; i<widgets.length; i++) {
-		
-		var zindex = doc.defaultView.getComputedStyle(widgets[i], null)
-			.getPropertyValue('z-index');
-		if (!zindex && zindex !== 0) continue;
-		
-		if (!topWidget || topZIndex < zindex) {
-			topWidget = widgets[i];
-			topZIndex = zindex;
-		}
-	}
-	
-	if (!topWidget) return false;
-	
-	var fields = topWidget.getElementsByClassName('field'),
-		found = false;
-	for (var i=0; i<fields.length; i++) {
-		var field = ZU.trimInternal(fields[i].textContent);
-		
-		var value = fields[i].nextElementSibling;
-		if (!value || !value.classList.contains('data')) continue;
-		value = ZU.unescapeHTML(ZU.cleanTags(value.innerHTML));
-		
-		if (!field || !value.trim()) continue;
-		
-		if (checkOnly) return true;
-		
-		found = true;
-		data[field] = value;
-	}
-	
-	return found ? data : false;
+	"lastUpdated": "2016-03-31 21:07:27"
 }
 
+/**
+    detectWeb is run to determine whether item metadata can indeed be retrieved from the webpage. 
+    The return value of this function should be the detected item type (e.g. “journalArticle”, 
+    see the overview of Zotero item types), or, if multiple items are found, “multiple”. 
+**/
 function detectWeb(doc, url) {
-	//monitor changes to body's direct children. That's where the metadata popup is added
-	Zotero.monitorDOMChanges(doc.body, {childList: true});
+    var itemType = false; // default - ignore
+    if (url.match(/\/iv2\.|ExternalIV.jsp/)) {
+        // Image viewer window
+        itemType = "artwork";
+    } else if (url.match(/\#3\|/)) {
+        // Thumbnail window page
+        if ((doc.getElementsByClassName('MetaDataWidgetRoot') != null) &&
```

Comment:
you can save yourself the first test. I'm pretty sure that function never returns null. It just returns an empty array where the class doesn't exists. Applies to several instances of this across the translator.